cypress/e2e/contacts.cy.js

```diff
@@ -0,0 +1,107 @@
+            cy.get(".moj-banner").should("contain", "Contact added");
+        });
+
+        it("shows error message when submitting empty data", () => {
+            cy.setCookie("fail-route", "addContactEmpty");
+            cy.get("#add-contact-form").submit();
+            cy.get(".govuk-error-summary__body").should("contain", "Enter a name");
+            cy.get(".govuk-error-summary__body").should("contain", "Enter a telephone number");
+            cy.get(".govuk-error-summary__body").should("contain", "Enter an email address");
+            cy.get(".govuk-error-summary__body").should("contain", "Select whether this contact is a main contact");
+            cy.get(".govuk-error-summary__body").should("contain", "Select whether this contact is the named deputy");
+        });
+
+        it("shows error message when submitting invalid data", () => {
+            cy.setCookie("fail-route", "addContactInvalid");
+            cy.get("#add-contact-form").submit();
+            cy.get(".govuk-error-summary__body").should("contain", "The name must be 255 characters or fewer");
+            cy.get(".govuk-error-summary__body").should("contain", "The job title must be 255 characters or fewer");
+            cy.get(".govuk-error-summary__body").should("contain", "Enter an email address in the correct format, like name@example.com");
+            cy.get(".govuk-error-summary__body").should("contain", "The telephone number must be 255 characters or fewer");
+            cy.get(".govuk-error-summary__body").should("contain", "The other telephone number must be 255 characters or fewer");
+            cy.get(".govuk-error-summary__body").should("contain", "The note must be 255 characters or fewer");
+        });
+    });
+
+    describe("List Contacts", () => {
+        beforeEach(() => {
+            cy.visit("/supervision/deputies/3/contacts");
+        })
+
+        it("shows header title and button", () => {
+            cy.get(".govuk-main-wrapper > header").contains("Contacts");
+            cy.get(".govuk-button").contains("Add new contact");
+        });
+
+        it("displays 4 column headings", () => {
+            cy.get(".govuk-table__row").find("th").should("have.length", 4);
+
+            const expected = [
+                "Contact",
+                "Contact details",
+                "Notes",
+                "Action",
+            ];
+
+            cy.get(".govuk-table__head > .govuk-table__row")
+                .children()
+                .each(($el, index) => {
+                    cy.wrap($el).should("contain", expected[index]);
+            });
+        });
+
+        it("should display contact data", () => {
+            cy.get(':nth-child(1) > :nth-child(1) > .name').contains("Minimal Contact");
+            cy.get(':nth-child(1) > :nth-child(2) > .email > a').contains("email@test.com");
+            cy.get(':nth-child(1) > :nth-child(2) > .phone-number').contains("0123456789");
```

Comment:
Just wondering if there's a cleaner way to get them, e.g. would `:nth-child(1) > .name` bring back that contact?